### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by jansepar**
## Changes
- Adds changes based on @johnboxall's Code Review talk

### CODE REVIEW

File: github-templates/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -15,7 +15,9 @@ Linked PRs: (links to corresponding PRs, optional)
 ## TODOs:
 - [ ] +1
 - [ ] Passes tests
+- [ ] "How to test" steps have been run through
```
Comment (kschmidtdev) [inline github-templates/PULL_REQUEST_TEMPLATE.md]: Might be implicitly understood here, but we would prefer that the testing was also successful :)

Comment (jeremywiebe) [inline github-templates/PULL_REQUEST_TEMPLATE.md]: I actually feel like this is a checklist for the author, not the reviewer.... so does this make sense here?  The assumption (as Karl points out) is that the reviewer uses the How to Test section and tests.

Comment (jansepar) [inline github-templates/PULL_REQUEST_TEMPLATE.md]: Ah, good point. Yeah the +1 essentially means that the reviewer has ran those steps. I'll remove.

**Comment by marlowpayne** [comment]
👍 :shipit:

**Pull request merged**